### PR TITLE
Implemented appointment calendar cards

### PR DESCRIPTION
**Pull request by jadenPete**
This PR implements calendar cards, which are displayed for each appointment in the calendar cell in which the appointment belongs. To demonstrate the feature, two appointments are automatically added to the calendar: one from 9 to 9:30 AM on the current day, and one from 9:30 AM to 10 AM on the current day. Eventually, when we prepopulate the database with dummy data, these appointments should be pulled dynamically from the database.

Screenshot 2024-03-05 at 15-33-05 Screenshot

To conserve space, only the name of the section to which the appointment belongs is shown. Then, when the appointment is hovered over, the name of the instructional member and the time of the appointment is shown.

Screenshot 2024-03-05 at 15-38-50 Screenshot

Additionally, this PR reconfigures the calendar to dynamically determine the start and end times according to the provided appointments, a minimum start time, and a maximum end time.

Note: Once we've achieved an MVP, I'd like to revisit this design and show a "+X appointments" button at the bottom of overflowed calendar cells, similar to how Google Calendar does it, instead of making cells scrollable.

### CODE REVIEW

File: src/lib/components/calendar/CalendarCard.svelte [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		</div>
+	</div>
+</div>
+
+<style>
+	* {
+		--animation-duration: 0.2s;
+	}
+
+	@keyframes calendar-card-section-in {
```
Comment (DylanHalstead) [inline src/lib/components/calendar/CalendarCard.svelte:48]: Did you look into using hover: and transition? With on 2 frames I would think this is a simpler approach
https://tailwindcss.com/docs/transition-property

**Review by DylanHalstead — APPROVED** [review]
LGTM

File: src/lib/components/calendar/CalendarCard.svelte (hunk 1 above)
Comment (jadenPete) [reply src/lib/components/calendar/CalendarCard.svelte:48]: It looks like Tailwind's `animation` and `transition` classes only support a limit set of properties.
My transition affects the `grid-template-rows` property, which Tailwind doesn't seem to support.
My animation affects the `font-size` property, which Tailwind also doesn't seem to support.

Thanks for the suggestion, though.

**Comment by jadenPete** [comment]
Rebased off of `main`.

**Comment by jadenPete** [comment]
Squashed.

**Pull request merged**